src/main/java/org/haffson/adventofcode/days/day02/Day02.java

```diff
@@ -0,0 +1,177 @@
+package org.haffson.adventofcode.days.day02;
+
+import org.haffson.adventofcode.ProblemStatusEnum;
+import org.haffson.adventofcode.days.Days;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Scanner;
+
+
+
+/**
+ * Implementation for <i>Day 1: Chronal Calibration</i>.
+ */
+@Component
+public class Day02 implements Days {
+
+    private final String path = "/Users/jenni/dedica/AdventOfCode/Jenni/Day02/src/input.txt";
+    private List<String> rawData = getRawDataAsList(path);
+
+    /** The puzzle status {@code HashMap} */
+    private final HashMap<String, ProblemStatusEnum> problemStatus;
+
+    @Autowired
+    Day02() {
+        this.problemStatus = new HashMap<>();
+        this.problemStatus.put("1", ProblemStatusEnum.SOLVED);
+        this.problemStatus.put("2", ProblemStatusEnum.SOLVED);
+    }
+
+    @Override
+    public int getDay() {
+        return 2;
+    }
+
+    @Override
+    public HashMap<String, ProblemStatusEnum> getProblemStatus() {
+        return problemStatus;
+    }
+
+    @Override
+    public String firstPart() {
+        try {
+            return "Part 1 answer: " + getNumberPwd1();
+        } catch (FileNotFoundException e) {
+            return "Error in method 1";
+        }
+    }
+
+    @Override
+    public String secondPart() {
+        try {
+            return "Part 2 answer: " + getNumberPwd2();
+        } catch (FileNotFoundException e) {
+            return "Error in method 2";
+        }
+    }
+
+
+    /** Method to read raw data from file into list
+     *
+     * @return raw data as list
+     */
+    public List<String> getRawDataAsList(String path) {
+        List<String> rawData = new ArrayList<>();
+
+        try {
+            Scanner s = new Scanner(new File(path)).useDelimiter("\n");
+            while (s.hasNext()) {
+                rawData.add(s.next());
+            }
+            s.close();
+        } catch (FileNotFoundException e) {
+            System.out.println("File not found!");
+        }
```

Comment:
Have a look at the `Scanner` class. It implements (among some others), `Closeable`. This allows us, to put the declaration of the `Scanner` object inside a so-called `try-with-resources` block:
```suggestion
       try (Scanner s = new Scanner(new File(path)).useDelimiter("\n")){
            while (s.hasNext()) {
                rawData.add(s.next());
            }
        } catch (FileNotFoundException e) {
            System.out.println("File not found!");
        }
```
This makes sure, that the resources will be closed, even if the exception occurs. See also: https://www.baeldung.com/java-try-with-resources